firestore/client.go

```diff
@@ -101,6 +101,22 @@ func NewClient(ctx context.Context, projectID string, opts ...option.ClientOptio
 	return c, nil
 }
 
+// NewClientWithDatabase creates a new Firestore client that accesses the
+// specified database.
+func NewClientWithDatabase(ctx context.Context, projectID string, databaseID string, opts ...option.ClientOption) (*Client, error) {
+	if databaseID == "" {
+		return nil, errors.New("firestore: databaseName was empty")
+	}
+
+	clientPtr, err := NewClient(ctx, projectID, opts...)
```

Comment:
nit: I would just call this var client. Most of the struct types you find with methods will have point receivers.

Reply:
Done.